src/rhub/api/_setup.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+import retrying
+
+from rhub.api import db, get_keycloak
+from rhub.auth import ADMIN_USER, ADMIN_GROUP, ADMIN_ROLE
+from rhub.tower import model as tower_model  # noqa: F401
+from rhub.lab import model as lab_model  # noqa: F401
+from rhub.policies import model as policies_model  # noqa: F401
+
+
+logger = logging.getLogger(__name__)
+
+
+# This function must be idempotent. In container, it may be called on every
+# start.
+@retrying.retry(stop_max_attempt_number=3, wait_fixed=5000)
+def setup():
+    db.create_all()
+
+    keycloak = get_keycloak()
+
+    groups = {group['name']: group for group in keycloak.group_list()}
+    roles = {role['name']: role for role in keycloak.role_list()}
+
+    if ADMIN_GROUP not in groups:
+        logger.info(f'Creating admin group "{ADMIN_GROUP}"')
+        groups[ADMIN_GROUP] = keycloak.group_create({'name': ADMIN_GROUP})
+
+    if ADMIN_ROLE not in roles:
+        logger.info(f'Creating admin role "{ADMIN_ROLE}"')
+        roles[ADMIN_ROLE] = keycloak.role_create({'name': ADMIN_ROLE})
+
+    if not any(role['name'] == ADMIN_ROLE
+               for role in keycloak.group_role_list(groups[ADMIN_GROUP]['id'])):
+        logger.info(f'Adding admin role "{ADMIN_ROLE}" to admin group "{ADMIN_GROUP}"')
+        keycloak.group_role_add(ADMIN_ROLE, groups[ADMIN_GROUP]['id'])
+
+    if not keycloak.user_list({'username': ADMIN_USER}):
+        logger.info(f'Creating admin account "{ADMIN_USER}"')
+        keycloak.user_create({
+            'username': ADMIN_USER,
+            'email': 'nobody@redhat.com',
+            'firstName': 'Admin',
+            'lastName': 'RHub',
+        })
+
+    admin_user = keycloak.user_list({'username': ADMIN_USER})[0]
+    if not any(group['name'] == ADMIN_GROUP
+               for group in keycloak.user_group_list(admin_user['id'])):
```

Comment:
Spacing seems off (or is that just because of GitHub)?

Reply:
What do you mean? It looks ok on GitHub and flake8 is not complaining.

Reply:
@fpob  this is strange, I just cloned fpob:fixes and it looks wrong there too.  It looks just like what @sai-c posted above.

Reply:
I have no idea what you see :smile: 

![strange-indent](https://user-images.githubusercontent.com/6289078/123637544-6fb9d400-d81e-11eb-8864-691fc268677f.png)

^ do you see something else?

Reply:
Oh, I actually do see the same thing, but I guess I missed that the for loop was inside the any().

Reply:
Ah, see it now, the 'for' statement is part of the 'if' statement...missed the colon at the end.